### PR TITLE
Disable workspace diagnostics for the razor client.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Review by TanayParikh — APPROVED** [review]

**Review by dibarbet — APPROVED** [review]

**Review by TanayParikh — COMMENTED** [review]
Just realized, this shouldn't be required for razor, as we can just not make the workspace diagnostics call to roslyn. The question is whether this is required for platform such that *razor specific* document diagnostics aren't reported directly from roslyn -> platform.

Question: how're C# docs within a workspace with razor & C# files handled in terms of workspace diagnostics?

**Comment by CyrusNajmabadi** [comment]
> Just realized, this shouldn't be required for razor, as we can just not make the workspace diagnostics call to roslyn.

That's a good point.  I can close this out. or i can even assert that this never gets called so that we can catch mistakes though. 

> The question is whether this is required for platform such that razor specific document diagnostics aren't reported directly from roslyn -> platform.

It depends.  if platform calls through this client, we might have an issue.

File: src/Features/LanguageServer/Protocol/Handler/Diagnostics/WorkspacePullDiagnosticHandler.cs [hunk 1]
```diff
@@ -40,6 +40,12 @@ protected override WorkspaceDiagnosticReport CreateReport(TextDocumentIdentifier
 
         protected override ImmutableArray<Document> GetOrderedDocuments(RequestContext context)
         {
+            // If we're being called from razor, we do not support WorkspaceDiagnostics at all.  For razor, workspace
+            // diagnostics will be handled by razor itself, which will operate by calling into Roslyn and asking for
+            // document-diagnostics instead.
+            if (context.ClientName != null)
```
Comment (davidwengier) [inline src/Features/LanguageServer/Protocol/Handler/Diagnostics/WorkspacePullDiagnosticHandler.cs:46]: At this point, I feel like we have enough checks for this sort of thing that having a constant for the actual Razor client name might just be easier :)

**Review by davidwengier — APPROVED** [review]

**Review by TanayParikh — APPROVED** [review]
Okay it looks like platform will still be requesting roslyn for workspace diagnostics, so this will in fact be required.

File: src/Features/LanguageServer/Protocol/Handler/Diagnostics/WorkspacePullDiagnosticHandler.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/LanguageServer/Protocol/Handler/Diagnostics/WorkspacePullDiagnosticHandler.cs:46]: agreed.  i think this all just boils down to a .IsRazor bit.

**Pull request merged**